rclcpp/src/rclcpp/publisher.cpp

```diff
@@ -170,7 +170,14 @@ PublisherBase::setup_intra_process(
   StoreMessageCallbackT callback,
   const rcl_publisher_options_t & intra_process_options)
 {
-  auto intra_process_topic_name = std::string(this->get_topic_name()) + "/_intra";
+  const char * topic_name = this->get_topic_name();
+  std::string intra_process_topic_name = "";
+  if (topic_name) {
+    intra_process_topic_name = std::string(topic_name) + "/_intra";
+  } else {
+    throw std::runtime_error("Get a invalid null topic name");
+  }
```

Comment:
Please use a similar structure here as above. Check for `!topic_name` and conditionally `throw`, Then the `else` case becomes obsolete.

Reply:
make sense, changed :smile: